'gopherjs test' mode ignores tags

It appears that tags are ignored by the 'test' mode.

Consider this `foo` package with two implementations:

`foo.go`:

    // +build foo

    package foo

    func Foo() string {
    	return "foo"
    }

`nonfoo.go`:

    // +build !foo

    package foo

    func Foo() string {
    	return "bar"
    }

And `foo_test.go`:

    package foo
    
    import (
    	"testing"
    )
    
    func TestFoo(t *testing.T) {
	if x := Foo(); x != "foo" {
    		t.Fatalf("Expected foo, got %s", x)
    	}
    }

Running gopherjs test, as normal, will fail as expected:

    $ gopherjs test
    --- FAIL: TestFoo (0.06s)
            jstest_test.go:9: Expected foo, got bar                                                                                                              
    FAIL
    FAIL    github.com/flimzy/jstest        0.429s

But specifying the 'foo' tag also fails:

    $ gopherjs test --tags=foo
    --- FAIL: TestFoo (0.06s)
            jstest_test.go:9: Expected foo, got bar                                                                                                              
    FAIL
    FAIL    github.com/flimzy/jstest        0.422s

In looking at `tool.go`, I see that `cmdTest` is never told about `flagTags`, which seems like a confirmation that this functionality is not implemented.  I haven't dug much deeper.

--- a/build/build.go
+++ b/build/build.go
@@ -121,8 +121,8 @@ func importWithSrcDir(path string, srcDir string, mode build.ImportMode, install
 
 // ImportDir is like Import but processes the Go package found in the named
 // directory.
-func ImportDir(dir string, mode build.ImportMode) (*PackageData, error) {
-	pkg, err := build.ImportDir(dir, mode)
+func ImportDir(dir string, mode build.ImportMode, installSuffix string, buildTags []string) (*PackageData, error) {
+	pkg, err := NewBuildContext(installSuffix, buildTags).ImportDir(dir, mode)
 	if err != nil {
 		return nil, err
 	}

--- a/tool.go
+++ b/tool.go
@@ -287,13 +287,15 @@ func main() {
 	outputFilename := cmdTest.Flags().StringP("output", "o", "", "Compile the test binary to the named file. The test still runs (unless -c is specified).")
 	cmdTest.Flags().AddFlag(flagMinify)
 	cmdTest.Flags().AddFlag(flagColor)
+	cmdTest.Flags().AddFlag(flagTags)
 	cmdTest.Run = func(cmd *cobra.Command, args []string) {
+		options.BuildTags = strings.Fields(*tags)
 		os.Exit(handleError(func() error {
 			pkgs := make([]*gbuild.PackageData, len(args))
 			for i, pkgPath := range args {
 				pkgPath = filepath.ToSlash(pkgPath)
 				var err error
-				pkgs[i], err = gbuild.Import(pkgPath, 0, "", nil)
+				pkgs[i], err = gbuild.Import(pkgPath, 0, "", options.BuildTags)
 				if err != nil {
 					return err
 				}
@@ -310,7 +312,7 @@ func main() {
 					if err != nil {
 						return err
 					}
-					if pkg, err = gbuild.Import(pkgPath, 0, "", nil); err != nil {
+					if pkg, err = gbuild.Import(pkgPath, 0, "", options.BuildTags); err != nil {
 						return err
 					}
 				}

--- a/tool.go
+++ b/tool.go
@@ -317,7 +317,7 @@ func main() {
 					}
 				}
 				if pkg == nil {
-					if pkg, err = gbuild.ImportDir(currentDirectory, 0); err != nil {
+					if pkg, err = gbuild.ImportDir(currentDirectory, 0, "", options.BuildTags); err != nil {
 						return err
 					}
 					pkg.ImportPath = "_" + currentDirectory

